File: content/Data Hiding/CS2 Python/decrypt.py
def decrypt(c, k):
	""" Shift cipher decryption. Call for each letter of ciphertext.
	
	Input:
	c - the ciphertext letter (upper case character)
	k - the key, the integer number of letters to shift by
	Output:
	p - the plaintext letter (lower case character)
	"""
	p = (ord(c) - k - ord('A')) % 26
	return chr(p + ord('a'));

File: content/Secure Coding/Data Hiding/CS2 C++/decrypt.cpp
/*
 * Shift cipher decryption. Call for each letter of ciphertext.
 * 
 * c - the ciphertext letter (upper case)
 * k - the key, the number of letters to shift by
 * return the plaintext letter  (lower case)
 */
char decrypt(char c, int k) {
	int p = (c - k - 'A') % 26;
	if(p < 0) p += 26;
	return (char)(p + 'a');
}
